wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextDsl.kt

```diff
@@ -0,0 +1,210 @@
+     *
+     * @author NoComment1105
+     */
+    public fun keyBind(key: String, action: MutableStyle.() -> Unit = { }) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.keyBind(key)))
+    }
+
+
+    /**
+     * @see Text.nbt
+     *
+     * @author NoComment1105
+     */
+    public fun nbt(
+        pathPattern: String,
+        interpreting: Boolean,
+        separator: Optional<Text>,
+        nbt: TextData,
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.nbt(pathPattern, interpreting, separator, nbt)))
+    }
+
+    /**
+     * @see Text.nbt
+     *
+     * @author NoComment1105
+     */
+    public fun nbt(
+        pathPattern: String,
+        interpreting: Boolean,
+        separator: Optional<Text>,
+        nbt: ((ServerCommandSource) -> Stream<NbtCompound>),
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.nbt(pathPattern, interpreting, separator, nbt)))
+    }
+
+    /**
+     * Add a plain text
+     *
+     * @param value The text to add
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public fun text(value: String, action: MutableStyle.() -> Unit = { }) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.empty().append(value)))
+    }
+
+    /**
+     * Adds a scoreboard
+     *
+     * @param name The name to add to the scoreboard
+     * @param objective The objective of the scoreboard
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public fun scoreboardText(
+        name: String,
+        objective: String,
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        text.append(mutableStyle.applyTo(Text.score(name, objective)))
+    }
+
+    /**
+     * Adds a resolvable entity selector
+     *
+     * @param separator the optional separator if there's multiple matches issued from the selector
+     * @param selector the selector
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public fun selectorText(
+        selector: String,
+        separator: Optional<Text>,
+        action: MutableStyle.() -> Unit = { }
+    ) {
+        val mutableStyle = MutableStyle()
+        mutableStyle.apply(action)
+        this.text.append(mutableStyle.applyTo(Text.selector(selector, separator)))
+    }
+
+    /**
+     * Adds a mutable empty text
+     *
+     * @see MutableStyle for action
+     *
+     * @author NoComment1105
+     */
+    public fun empty(action: MutableStyle.() -> Unit = { }) {
```

Comment:
Whats the purpose of this method? is there an actual reason to ever append an empty mutable text?

Reply:
It's an available method in Minecraft code, so we should provide something and allow formatting because end users may want to do some odd stuff

Reply:
...bizarre, but it is Mojang, haha